### DISCUSSION TITLE
The memory used 100% in cluter smaller memory nodes

### DISCUSSION THREAD

**Discussion by aliuq** [k3s-io/k3s#5988] [Q&A]
Hi, I tried setting kubelet-arg with `system-reserved=cpu=100m,memory=800Mi,ephemeral-storage=1Gi`, but it seems not works, I hope when the smaller memory nodes reaches 3.2g and it should keep works

The node args

```yaml
k3s.io/node-args: '["agent","--node-name","node-1","--docker","--node-external-ip","xx.xx.xx.xx", \
"--kube-proxy-arg","metrics-bind-address=0.0.0.0", \
"--kubelet-arg","system-reserved=cpu=100m,memory=800Mi,ephemeral-storage=1Gi"]'
```

![image](https://user-images.githubusercontent.com/11707934/184521750-927ea523-61ec-4da6-8066-e49e847b2045.png)

**Comment by brandond** [comment]
Setting the reserved resources just subtracts those amounts from the resources available to be scheduled for pods. It does not actually prevent your pods from consuming those resources. You should set appropriate resource requests and limits on your pods so that they do not consume excessive memory or CPU.

None of this is K3s specific, it's just how Kubernetes works.

**Reply by aliuq** [reply]
I think `--eviction-hard` will be resolved this question, I guessed.

```yaml
k3s.io/node-args: '["agent","--node-name","node-1","--docker","--node-external-ip","xxx.xxx.xxx.xxx", \
"--kube-proxy-arg","metrics-bind-address=0.0.0.0","--kubelet-arg", \
"system-reserved=cpu=100m,memory=800Mi,ephemeral-storage=1Gi","--kubelet-arg", \
"--eviction-hard","memory.available\u003c800Mi"]'
```

**Reply by brandond** [reply]
You still need to provide requests or at least limits, or containerized processes will be allowed to consume unlimited amounts of memory. No tinkering with the kubelet args will change that.

**Reply by aliuq** [reply]
Thank you for your reply, I will provide requests limits
